parsec/remote_dep_mpi.c

```diff
@@ -2152,67 +1692,105 @@ remote_dep_mpi_put_start(parsec_execution_stream_t* es,
         assert(k < MAX_PARAM_COUNT);
         if(!((1U<<k) & task->output_mask)) continue;
 
-        if(parsec_comm_puts == parsec_comm_puts_max) {
-            PARSEC_DEBUG_VERBOSE(20, parsec_comm_output_stream, "MPI:\treach PUT limit for deps 0x%lx. Reschedule.", deps);
-            parsec_list_nolock_push_front(&dep_put_fifo, (parsec_list_item_t*)item);
-            return;
-        }
-        PARSEC_DEBUG_VERBOSE(20, parsec_comm_output_stream, "MPI:\t[idx %d mask(0x%x / 0x%x)] %p, %p", k, (1U<<k), task->output_mask,
+        PARSEC_DEBUG_VERBOSE(20, parsec_debug_output, "MPI:\t[idx %d mask(0x%x / 0x%x)] %p, %p", k, (1U<<k), task->output_mask,
                 deps->output[k].data.data, PARSEC_DATA_COPY_GET_PTR(deps->output[k].data.data));
         dataptr = PARSEC_DATA_COPY_GET_PTR(deps->output[k].data.data);
         dtt     = deps->output[k].data.remote.src_datatype;
         nbdtt   = deps->output[k].data.remote.src_count;
+        (void) nbdtt;
+
+        task->output_mask ^= (1U<<k);
+
+        parsec_ce_mem_reg_handle_t source_memory_handle;
+        size_t source_memory_handle_size;
+
+        if(parsec_ce.capabilites.supports_noncontiguous_datatype) {
+            parsec_ce.mem_register(dataptr, PARSEC_MEM_TYPE_NONCONTIGUOUS,
+                                   nbdtt, dtt,
+                                   -1,
+                                   &source_memory_handle, &source_memory_handle_size);
+        } else {
+            /* TODO: Implement converter to pack and unpack */
+            int dtt_size;
+            parsec_type_size(dtt, &dtt_size);
+            parsec_ce.mem_register(dataptr, PARSEC_MEM_TYPE_CONTIGUOUS,
+                                   -1, NULL, // TODO JS: this interface is so broken, fix it!
+                                   dtt_size, // TODO JS: what about nbdtt? Is it ok to ignore it?!
+                                   &source_memory_handle, &source_memory_handle_size);
+
+        }
+
+        parsec_ce_mem_reg_handle_t remote_memory_handle = item->cmd.activate.remote_memory_handle;
+
 #if defined(PARSEC_DEBUG_NOISIER)
         MPI_Type_get_name(dtt, type_name, &len);
```

Comment:
Do we have a parsec-level replacement for `MPI_Type_get_name`?

Reply:
nope, but maybe we need one.

Reply:
I think in my branch it's just surrounded by `#if defined(PARSEC_HAVE_MPI)`.